### PR TITLE
[4.0] Add prepared statements for plg_content_joomla

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
Use the plugin in all ways you can think of.

* Create an article in a category
* try to delete the category, you should get an error that the category is not empty
* create an article in the frontend, you should get a e-mail notification

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/content/joomla/joomla.php [hunk 1]
```diff
@@ -646,18 +648,20 @@ public function onContentChangeState($context, $pks, $value)
...
 				->from($db->quoteName('#__workflow_transitions', 't'))
 				->from($db->quoteName('#__workflow_stages', 's'))
-				->where($db->quoteName('t.from_stage_id') . ' = ' . (int) $assoc->stage_id)
+				->where($db->quoteName('t.from_stage_id') . ' = :stageid')
 				->where($db->quoteName('t.to_stage_id') . ' = ' . $db->quoteName('s.id'))
 				->where($db->quoteName('t.published') . '= 1')
 				->where($db->quoteName('s.published') . '= 1')
-				->order($db->quoteName('t.ordering'));
+				->order($db->quoteName('t.ordering'))
+				->bind(':stageid', $stageId);
```
Comment (Quy) [inline plugins/content/joomla/joomla.php]:
```suggestion
				->bind(':stageid', $stageId, ParameterType::INTEGER);
```

**Review by alikon — APPROVED** [review]

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f20a3412394d3fa1b00d6629aed0f166a5eff180<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25050">issues.joomla.org/tracker/joomla-cms/25050</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on f20a3412394d3fa1b00d6629aed0f166a5eff180<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25050">issues.joomla.org/tracker/joomla-cms/25050</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by roland-d** [comment]
Thank you.

**Pull request merged**